Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -603,6 +603,57 @@ extension PurchasesOrchestrator: StoreKit1WrapperDelegate {
 
 }
 
+extension PurchasesOrchestrator: PaymentQueueWrapperDelegate {
+
+    func paymentQueueWrapper(
+        _ wrapper: PaymentQueueWrapper,
+        shouldAddStorePayment payment: SKPayment,
+        for product: SK1Product
+    ) -> Bool {
+        // `PurchasesOrchestrator` becomes `PaymentQueueWrapperDelegate` only
+        // when `StoreKit1Wrapper` is not initialized, which means that promoted purchases
+        // need to be handled as a SK2 purchase.
+        // This method converts the `SKPayment` into an SK2 purchase by fetching the product again.
+        assert(self.storeKit1Wrapper == nil, "This method should not be invoked if SK1 is enabled")
+
+        guard let delegate = self.delegate else { return false }
+
+        let productIdentifier = product.productIdentifier
+
+        self.productsManager.products(withIdentifiers: [productIdentifier]) { result in
+            guard let product = result.value?.first(where: { $0.productIdentifier == productIdentifier }) else {
+                Logger.warn(Strings.purchase.promo_purchase_product_not_found(productIdentifier: productIdentifier))
+                return
+            }
+
+            let startPurchase: StartPurchaseBlock
+
+            if #available(iOS 12.2, macOS 10.14.4, watchOS 6.2, macCatalyst 13.0, tvOS 12.2, *),
+               let discount = payment.paymentDiscount.map(PromotionalOffer.SignedData.init) {
+                startPurchase = { completion in
+                    self.purchase(product: product,
+                                  package: nil,
+                                  promotionalOffer: discount) { transaction, customerInfo, error, cancelled in
+                        completion(transaction, customerInfo, error, cancelled)
+                    }
+                }
+            } else {
+                startPurchase = { completion in
+                    self.purchase(product: product,
+                                  package: nil) { transaction, customerInfo, error, cancelled in
+                        completion(transaction, customerInfo, error, cancelled)
+                    }
+                }
+            }
+
+            delegate.readyForPromotedProduct(product, purchase: startPurchase)
+        }
+
+        return false
```

Comment:
Looks like we always return false in this function. Also, I'm not sure what the returned boolean is supposed to mean... Maybe store in a variable with a meaningful name so it's easier to understand the returned value?

Reply:
Yeah that confused me to in the old implementation. Let me dig into why and document it 👍🏻

Reply:
Done.
I added this note:
> // See `SKPaymentTransactionObserver.paymentQueue(_:shouldAddStorePayment:for:)`
// Returns `false` to indicate that the app will defer the purchase and be handled
// when the user calls the purchase callback.